lib/livebook/content_loader.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Livebook.ContentLoader do
+  @moduledoc false
+
+  @doc """
+  Rewrite known URLs, so that they point to plain text file rather than HTML.
+
+  Currently the rewerites handle:
+
+    * GitHub files
+    * Gist files
+  """
+  @spec rewrite_url(String.t()) :: String.t()
+  def rewrite_url(url) do
+    url
+    |> URI.parse()
+    |> do_rewrite_url()
+    |> URI.to_string()
+  end
+
+  defp do_rewrite_url(%URI{host: "github.com"} = uri) do
+    case String.split(uri.path, "/") do
+      ["", owner, repo, "blob", hash | file_path] ->
+        path = Enum.join(["", owner, repo, hash | file_path], "/")
+
+        %{
+          uri
+          | path: path,
+            host: "raw.githubusercontent.com",
```

Comment:
While it is unlikely to change, the URL GitHub actually uses on the interface is https://github.com/elixir-nx/livebook/raw/main/.formatter.exs (note raw). We should probably use the same.

Reply:
That’s a good question too: does the current code handle redirects? Probably worth a test too that does two bypass requests.

Reply:
I don't think we need to support redirects, because in pretty much all cases the user would copy the URL from the given page, so it shouldn't result in redirect (well it could if the page requires authorization, but then it doesn't make sense to follow the redirect either). Any other scenario?